src/bcrypt.rs

```diff
@@ -13,12 +13,12 @@ fn setup(cost: u32, salt: &[u8], key: &[u8]) -> Blowfish {
     state
 }
 
-pub fn bcrypt(cost: u32, salt: &[u8], password: &[u8], output: &mut [u8]) {
-    assert!(salt.len() == 16);
+pub fn bcrypt(cost: u32, salt: [u8; 16], password: &[u8]) -> [u8; 24] {
```

Comment:
consider referencing salt instead giving ownership? because further in code we are still taking a reference